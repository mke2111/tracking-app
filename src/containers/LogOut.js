/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import { useDispatch } from 'react-redux';
import { resetData } from '../actions';

const LogOut = () => {
  const dispatch = useDispatch();
  const handleLogOut = () => {
    dispatch(resetData());
    localStorage.removeItem('token');
  };
  return (
    <div
      className="bg-red-500 p-3 cursor-pointer"
      onClick={handleLogOut}
      to="/"
    >
      <small>Log Out</small>
    </div>
  );
};

export default LogOut;
