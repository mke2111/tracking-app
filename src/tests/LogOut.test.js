import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../reducers/store';
import LogOut from '../containers/LogOut';

describe('Rendering component', () => {
  it('creates a LogOut component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <LogOut />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('Log Out');
  });
});

describe('LogOut', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <LogOut />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
