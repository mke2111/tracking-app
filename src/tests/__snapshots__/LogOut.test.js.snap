// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogOut renders correctly 1`] = `
<div
  className="bg-red-500 p-3 cursor-pointer"
  onClick={[Function]}
  to="/"
>
  <small>
    Log Out
  </small>
</div>
`;
